### ISSUE TITLE
Capabilities 2022

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/almanac.httparchive.org#2894]
# Capabilities 2022

![Capabilities illustration](https://raw.githubusercontent.com/HTTPArchive/almanac.httparchive.org/main/src/static/images/2020/capabilities/hero_lg.jpg)

If you're interested in contributing to the Capabilities chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @MichaelSolati | @MichaelSolati | @webmaxru @hemanth @beaufortfrancois @tomayac @christianliebel | @jrstanley | - | @foxdavidj |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [ ] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Capabilities resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-capabilities on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1-0dZd1Skd3PU20YtDn9rYEaysyLmCdi70s_hkyRdBFc/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/capabilities
[~google-sheets]: https://docs.google.com/spreadsheets/d/13S9FRj8OPRtoMPb94jFh6pPNz3lNS9yztIaorZYe288/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/capabilities.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

**Comment by MichaelSolati** [comment]
I'd definitely be interested in helping here. This will be my first time doing something like this, so I'd hope someone would be able to take lead. But I'm excited to help author this!

**Comment by webmaxru** [comment]
Hello!
Please, count me in as a Reviewer!

**Comment by hemanth** [comment]
Count me in on this! :)

**Comment by tomayac** [comment]
If additional reviewers are needed, I'm happy to help out as well.

**Comment by tomayac** [comment]
Note that we should now have _significantly more_ data, the [queries from 2021](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2021/capabilities) should all now deliver better results than what we had back then.

**Comment by beaufortfrancois** [comment]
I'll be happy to review content as well when needed.

**Comment by foxdavidj** [comment]
Hey @MichaelSolati, would you be interested in taking the [Chapter Lead role](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide) for the Capabilities chapter?

It sounds like you had some reservations, but it looks like we have a stellar team of reviewers from past years that would be able to help support you

**Comment by MichaelSolati** [comment]
> Hey @MichaelSolati, would you be interested in taking the [Chapter Lead role](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide) for the Capabilities chapter?
> 
> It sounds like you had some reservations, but it looks like we have a stellar team of reviewers from past years that would be able to help support you

Haha, it looks like I don't have much of a choice, so yeah I'll take lead.

**Comment by christianliebel** [comment]
@MichaelSolati I'm also happy to assist you. As the author and content team leader for the last two editions, I could help review the chapter or provide insight into how we structured the chapter in the previous two years… whatever you need!

**Comment by foxdavidj** [comment]
@MichaelSolati @webmaxru @hemanth @beaufortfrancois @tomayac @christianliebel

Hey everyone, excited to see we've got a full team (aside from an analyst) for the chapter!

To kick things off, I'd love to set up a 30 minute call within the next two weeks to put any new faces to names, and start the planning and brainstorming process.

@MichaelSolati as the Chapter Lead can you assist in finding a time that works for everyone? You can see my availability via my calendly here: https://calendly.com/foxdavid/30min

Also, here is an agenda for what we might want to discuss on the kickoff call: https://docs.google.com/document/d/1grj1XDilUg5VXgFpYk25TbIzrmma0B8XIjXTp5NfeLU/edit?usp=sharing

**Comment by jrstanley** [comment]
I would like to volunteer as an analyst for this chapter. This is my first time contributing, however I would like to help where I can!

**Comment by MichaelSolati** [comment]
> I would like to volunteer as an analyst for this chapter. This is my first time contributing, however I would like to help where I can!

Hey JR. If you go to my profile you should see my email address. Send me an email so I can loop you into the scheduling for our first meeting.

**Comment by rviscomi** [comment]
Welcome @jrstanley thanks for stepping up! Join us in the [`#web-almanac-analysts`](https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA) channel on Slack.

@MichaelSolati now that you've got a full content team, I've checked off Milestone 0, but I'll leave it up to you to keep the checklist up to date as you complete the milestones.

@MichaelSolati @webmaxru @hemanth @beaufortfrancois @tomayac @christianliebel @jrstanley could you all request access to the [chapter planning doc](https://docs.google.com/document/d/1-0dZd1Skd3PU20YtDn9rYEaysyLmCdi70s_hkyRdBFc/edit?usp=sharing) and start working towards Milestone 1, which is to complete the chapter outline by May 15? We've also got a [`#web-almanac-capabilities`](https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA) channel on Slack where you can coordinate on your chapter as needed.

Thanks everyone!

**Comment by foxdavidj** [comment]
@MichaelSolati Have you been able to find a time that works for the team to have a kickstart meeting? Sounds like you may have taken care of this via email or slack.

**Comment by MichaelSolati** [comment]
Hey @foxdavidj i just booked a time with you that worked for as many of the team as possible.

**Comment by foxdavidj** [comment]
@MichaelSolati, Heads-up that we're fast approaching the May 15th deadline of having an [outline for the chapter](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content) put together.

Instead of waiting for the kick-off meeting, it'd be a good idea to start working on the outline with the team. Especially if there are any new [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) the chapter requires.

**Comment by foxdavidj** [comment]
@MichaelSolati How is the outline for the chapter coming along. We're passed the May 15th deadline, and quickly heading running out of time to write any new custom metrics for the chapter.

**Comment by foxdavidj** [comment]
@MichaelSolati friendly ping

**Comment by MichaelSolati** [comment]
Sorry I was sick and kinda fell of the face of the earth for a bit. I'm responding to your email now.

**Comment by MichaelSolati** [comment]
@jrstanley I just sent you an email, take a look and let me know what you think.

**Comment by foxdavidj** [comment]
@MichaelSolati Have you and @jrstanley been able to connect offline to work on what metrics need to be queried?

Ideally, to give as much time as possible for writing and editing, we'd like to have all of the queries written by the end of the month. How are you feeling about meeting that deadline?

**Comment by MichaelSolati** [comment]
@foxdavidj we were able to identify what needs to be queried.

Im just waiting on @jrstanley to confirm the queries have been written.

**Comment by foxdavidj** [comment]
@jrstanley Could you create a draft PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)), that simply lists all of the queries in checkbox form? Makes it much easier to track the progress and see if there's anything we can hop in to help you with

**Comment by foxdavidj** [comment]
@MichaelSolati Can you reach out to @jrstanley ? Looks like the draft PR has still not been completed

**Comment by foxdavidj** [comment]
@MichaelSolati Can you give us a list of the metrics you were hoping to query? So we can help you find an analyst to start writing the queries?

Just in case @jrstanley isn't able to come back and assist

**Comment by foxdavidj** [comment]
@MichaelSolati friendly ping on the above

**Comment by MichaelSolati** [comment]
@foxdavidj 

- Query the top 20 used APIs (we'll pick 7-10 of them to discuss)
- Compare the usage of the APIs discussed last year against their usage this year (see list below)
- Compare the growth/loss of APIs selected to discuss this year against their usage last year

Separately here's a list of the APIs discussed last year:

- Async Clipboard API
- File System Access API
- Web Share API
- URL Handlers and Declarative Link Capturing
  - URL Handling
  - Declarative Link Capturing
- Web USB API
- Web Bluetooth API
- Web Serial API
- Generic Sensor API

**Comment by tomayac** [comment]
Just two things to point out:

>   * URL Handling

From https://web.dev/pwa-url-handler/: 

_"PWAs as URL Handlers was part of the [capabilities project](https://developer.chrome.com/blog/fugu-status/) and support for the experimental url_handlers manifest member, documented below, is being phased out. The url_handlers manifest member is being replaced by the new handle_links manifest member, which is currently being standardized and implemented. See the [handle_links explainer](https://github.com/WICG/pwa-url-handler/blob/main/handle_links/explainer.md)."_

>   * Declarative Link Capturing

From https://developer.chrome.com/docs/web-platform/declarative-link-capturing/:

_"Declarative Link Capturing was part of the [capabilities project](https://developer.chrome.com/blog/fugu-status/). The engineering team has decided that Declarative Link Capturing will not launch with its current design. Instead, the feature has been redesigned as described in [Control how your app is launched](https://developer.chrome.com/docs/launch-handler/). If your app implements the Declarative Link Capturing API, you should [transition to the replacement Launch Handler API](https://developer.chrome.com/docs/web-platform/declarative-link-capturing/#migration)."_

> * Web USB API
> * Web Bluetooth API
> * Web Serial API

If we have data, can we additionally discuss WebHID and WebNFC?

**Comment by MichaelSolati** [comment]
I'd be open to highlighting WebHID and WebNFC. I wouldn't want to shove it in, though, so I'll have to think about how to frame it with our outline.

**Comment by foxdavidj** [comment]
Hey @MichaelSolati, it looks like the analysis portion for the chapter has really fallen behind (looks like @jrstanley may have had something come up)... and we're now eating into the time allotted for writing the chapter.

I talked about the issue with other members of the team and while I haven't been able to find a replacement analyst, I did get some volunteers who’d be down to take last year’s queries, update, and re-run all of them with this year’s data.

You wouldn’t have any new metrics, but you would have the same data that was used in last year’s chapter.

Is this something you’d be interested in to continue writing the chapter?

**Comment by MichaelSolati** [comment]
@foxdavidj, that's a little unfortunate, but I don't mind. As long as we output a meaningful entry, I'll be happy. I'm sure we'll be able to.

**Comment by tunetheweb** [comment]
Hi all,

There's actually only two queries that were run for last year, so I've updated that and rerun it for this year. The results are now available in the [2022 Capabilities results sheet](https://docs.google.com/spreadsheets/d/13S9FRj8OPRtoMPb94jFh6pPNz3lNS9yztIaorZYe288/edit#gid=871711070).

The good news, is that this grabs ALL the Fugu API information we have, and @tomayac has been busy updating the [fugu custom metric](https://github.com/HTTPArchive/custom-metrics/blob/main/dist/fugu-apis.js) over the last year, so we actually have data already on some new APIs in these results that we didn't have last year! The Launch Handler API is one that we didn't have last year to give one example. So hopefully there's still some good info in there.

For [last year's chapter](https://almanac.httparchive.org/en/2021/capabilities) a lot of stats were given, just from this table as single stats. However, if you can think of any other ways of presenting the data then let us know and can look at adding visualisations. A few [trend lines graphs](https://almanac.httparchive.org/en/2021/capabilities#fig-3) were obtained from Chrome Status and copying the data from that, so again we can do similar this year. See [last year's results sheet](https://docs.google.com/spreadsheets/d/1b4moteB9EiLYkH1Ln9qfi1tnU-E4N2UQ87uayWytDKw/edit#gid=1382342903) for the two examples of that.

@christianliebel also added a lot of screenshots to last years chapter, but he got those himself, rather than the analyst so feel free to do similar yourself.

Anyway, I think you have the data to unblock you now, but do reach out if there's any more help you need. I don't know the subject at all but can help with any BigQuery or any more generic Web Almanac questions. And between that, and your reviewers, hopefully we can get an interesting chapter on the subject!

**Comment by foxdavidj** [comment]
@MichaelSolati How's the chapter coming along?

**Comment by foxdavidj** [comment]
Hi @MichaelSolati hope you're doing well. Looks like there isnt any activity in the Google Doc and the writing deadline is fast approaching.

Is this something you'll be able to commit to writing in the next couple weeks? Because, sadly, right now this chapter is at risk of being closed so that we can focus our effort on the other chapters.

**Comment by MichaelSolati** [comment]
@foxdavidj 



I'll be going to work on it over the weekend. I'm not going to let this be closed 💪

**Comment by MichaelSolati** [comment]
@foxdavidj the chapter is maybe about 1/3 of the way complete. I'll give another update tomorrow as I make more progress. I'll limit the chapter to 10 capabilities as it could go on for a while; however, I'm pleased with what I've written.

@webmaxru @hemanth @beaufortfrancois @tomayac @christianliebel I invite you to review what's been written so far. You can find the article here => https://docs.google.com/document/d/1-0dZd1Skd3PU20YtDn9rYEaysyLmCdi70s_hkyRdBFc/edit#heading=h.2intlsfa36tg

I'll ping y'all here with updates as I make them.

**Comment by MichaelSolati** [comment]
@foxdavidj, I'm going to say it's at 60% completion.

The last API I'll be writing about is the Push API. After Push, no other API we scraped for was used by more than 2% of the pages. I'm thinking 2% is a good cutting-off point. 

I will also write a section about Fugu and a conclusion. I think this will be the bare minimum to have a solid chapter. 

My first "stretch" goal for the next day is to get screenshots of sites implementing the features shown. The second stretch goal is to write a couple of sentences about each of the screenshots and the sites using them.

I say stretches, I expect them to be ready before publishing/review, but they're lower priority than Push, Fugu, and the conclusion.

@tomayac and @hemanth, there are four new sections, Media Session, Device Memory, Service Worker, and Gamepad. Please review them at your earliest convenience.

https://docs.google.com/document/d/1-0dZd1Skd3PU20YtDn9rYEaysyLmCdi70s_hkyRdBFc/edit#heading=h.2intlsfa36tg

**Comment by foxdavidj** [comment]
Great to hear @MichaelSolati . And looks like the reviewers have already hopped in with some comments

**Comment by foxdavidj** [comment]
Hey @MichaelSolati, we're past the September 15th deadline so i really want to get this chapter sent off to editing ASAP.

How confident are you in being able to address the reviews and tackling the conclusion by EOD Monday, September 19?

**Comment by MichaelSolati** [comment]
@foxdavidj I'll be able to have things done by EOD 9/19/22.

**Comment by MichaelSolati** [comment]
@foxdavidj Capabilities draft is done, and I've made changes based on the feedback I've received thus far.

I've also opened a pr, #3099, with the content from the draft, which will be updated based on changes that may need to be made.

The only question's remaining are:

1. I don't have any featured quotes or stats; is that something you'd like me to fill out? Or is this something that can/would be decided by someone else?
2. We had some people volunteer to partake in this year's chapter who haven't made any contributions. Is it safe to remove them from the chapter?
3. I'm not sure what my next steps are; what else do I need to be prepared to do?

**Comment by foxdavidj** [comment]
@MichaelSolati 

1. Yes please do if there were some that really stood out to you! Would be a massive help
2. Yes. It's safe to remove them
3. Looks like editors have already stepped into your PR and made some changes. Only thing left for you would be to monitor the PR in case they have any questions

🎉

**Comment by mikaelaraujo** [comment]
@hemanth @beaufortfrancois @tomayac @christianliebel  @jrstanley We want to invite you to participate of [2025 edition of Capabilities chapter](https://github.com/HTTPArchive/almanac.httparchive.org/issues/4065#issuecomment-2894295589)!

**Comment by christianliebel** [comment]
@mikaelaraujo Thanks for the request. Unfortunately, I can't support the Capabilities chapter this year, as I have already taken the lead for the GenAI chapter.

**Comment by mikaelaraujo** [comment]
> [@mikaelaraujo](https://github.com/mikaelaraujo) Thanks for the request. Unfortunately, I can't support the Capabilities chapter this year, as I have already taken the lead for the GenAI chapter.

No worries @christianliebel! GenAI chapter will be incredible!
